### ISSUE TITLE
in the Find excel 2013 Office Box It is not possible to walk letter by letter

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#8973]
### Steps to reproduce:
open excel 2013.
Enter any content in some cells.
Press the command to open the find and replace box, in English Brazil the shortcut is ctrl + l.
Enter a text and press the left and right arrows to read character by character.
### Actual behavior:
The nvda always repeats the first letter of the text that was typed.

### Expected behavior:
nvda should read character by character of the text entered in the search box.
#### NVDA Installed/portable/running from source:
install
#### NVDA version:
alfa 16298
#### Windows version:
10 1717763.134
#### Name and version of other software in use when reproducing the issue:
excel 2013
#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA?
no

**Comment by comanna** [comment]
The same behaviour in Excel 2016. 

 

All the best,

 

Cearbhall

 

m +353 (0)833323487 Ph: _353 (0)1-2864623 e: cearbhall.omeadhra@blbc.ie

**Comment by Adriani90** [comment]
I can reproduce this in MS Excel 2016 and MS Office 365

**Comment by DrSooom** [comment]
Please press TAB followed by Shift+TAB after opening the search window with CTRL+F. In Microsoft Excel 2010 NVDA 2018.1.1 on Win7x64 is completely silent because it seems that NVDA isn't able to recognized the text field on the very first appearance of the find window.

**Comment by fernando-jose-silva** [comment]
in office 2013 and 2016 tabe + shift + tabe does not improve.
Thank you all.

**Comment by Adriani90** [comment]
@fernando-jose-silva, @lukaszgo1 I think this issue is solved in last Alpha after merging #9063. Right?

**Comment by lukaszgo1** [comment]
I do not have access to Excel 2013, therefore I cannot say for certain, but it should be fixed. We should wait for confirmation from someone using Excel 2013 before closing it though.

**Comment by fernando-jose-silva** [comment]
I no longer have access to office 2013 to test, however:
Testing with office 2019 last nvda alpha build and windows 10 17763 got the results:
Pressing the shortcut key to open the Find Nvda window did not announce that the search window was open.
When typing characters, nvda did not announce the characters entered.
When trying to read character by character the nvda was mute.
By pressing alt + tab and alt + tab again for excel nvda announced the search window, announced the text field and announced the characters entered, however when trying to read character by character, nvda always announces the first letter of the typed text and not read all the characters present in the text field.

**Comment by lukaszgo1** [comment]
@fernando-jose-silva Can you please paste developer info for that control?
To do this:
1. Open Excel find dialog
2. Ensure that the focus is in the edit field.
3. Move navigator object to focus.
4. press NVDA+f1, select all and paste it here.
Thanks

**Comment by Adriani90** [comment]
Hmm this is strange. In NVDA last Alpha, when I press ctrl+f in MS Excel, NVDA does not report anything like in Fernando’s case. But after pressing alt+tag I can navigate letter by letter. Here is the log after pressing ctrl+f:

 

```

IO - inputCore.InputManager.executeGesture (22:52:48.503):

Input: kb(laptop):control+f

IO - speech.speak (22:52:49.111):

Speaking [u'Suchen und Ersetzen  Dialogfeld']

IO - inputCore.InputManager.executeGesture (22:52:50.227):

Input: kb(laptop):shift+h

IO - inputCore.InputManager.executeGesture (22:52:50.549):

Input: kb(laptop):e

IO - inputCore.InputManager.executeGesture (22:52:50.832):

Input: kb(laptop):l

IO - inputCore.InputManager.executeGesture (22:52:50.967):

Input: kb(laptop):l

IO - inputCore.InputManager.executeGesture (22:52:51.138):

Input: kb(laptop):o

IO - inputCore.InputManager.executeGesture (22:52:54.279):

Input: kb(laptop):NVDA+f1

INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (22:52:54.957):

Developer info for navigator object:

name: u'Suchen und Ersetzen'

role: ROLE_DIALOG

states: STATE_FOCUSABLE

isFocusable: True

hasFocus: False

Python object: <NVDAObjects.Dynamic_DialogSDMIAccessibleWindowNVDAObject object at 0x05869ED0>

Python class mro: (<class 'NVDAObjects.Dynamic_DialogSDMIAccessibleWindowNVDAObject'>, <class 'NVDAObjects.behaviors.Dialog'>, <class 'NVDAObjects.IAccessible.msOffice.SDM'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)

description: ''

location: RectLTWH(left=1431, top=1156, width=1184, height=453)

value: None

appModule: <'excel' (appName u'excel', process ID 6832) at address 58697d0>

appModule.productName: u'Microsoft Office'

appModule.productVersion: u'16.0.11126.20266'

TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>

windowHandle: 264560

windowClassName: u'bosa_sdm_XL9'

windowControlID: 0

windowStyle: -1798569984

windowThreadID: 11712

windowText: u'Suchen und Ersetzen'

displayText: u'SuchenErsetzen\nHelloSuchen nach:\nOptionen >>\nAlle suchen Weitersuchen Schlie\xdfen'

IAccessibleObject: <POINTER(IAccessible) ptr=0x82b24e0 at 5872ad0>

IAccessibleChildID: 0

IAccessible event parameters: windowHandle=264560, objectID=-4, childID=0

IAccessible accName: u'Suchen und Ersetzen'

IAccessible accRole: ROLE_SYSTEM_DIALOG

IAccessible accState: STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048576)

IAccessible accDescription: None

IAccessible accValue: exception: (-2147352573, 'Mitglied nicht gefunden.', (None, None, None, 0, None))

```

 

The exception in IAccessible means (member not found), translated from german.

 

 

 

Von: Łukasz Golonka <notifications@github.com> 
Gesendet: Montag, 4. Februar 2019 22:51
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Comment <comment@noreply.github.com>
Betreff: Re: [nvaccess/nvda] in the Find excel 2013 Office Box It is not possible to walk letter by letter (#8973)

 

@fernando-jose-silva <https://github.com/fernando-jose-silva>  Can you please paste developer info for that control?
To do this:

1.	Open Excel find dialog
2.	Ensure that the focus is in the edit field.
3.	Move navigator object to focus.
4.	press NVDA+f1, select all and paste it here.
Thanks

**Comment by fernando-jose-silva** [comment]
Thank you very much, follow the log.
I believe that I made a mistake when I was describing, nvda announces the characters when typing, but does not advertise when navigating with the left and right arrows, as well as not announcing the window name and the text field when pressing the shortcut to open the search window in excel.
[nvda_excel.txt](https://github.com/nvaccess/nvda/files/2830214/nvda_excel.txt)

**Comment by Adriani90** [comment]
Fernando which alpha build are you using?

Von meinem iPhone gesendet

**Comment by leonardder** [comment]
I can confirm that NVDA doesn't report focus on the edit box when opening the search dialog for the first time. I can also reproduce this with narrator. These edit fields are really broken. #9063 improves things, though.

@dkager: could you have a look with Excel 2013?

**Comment by dkager** [comment]
Yes, I can reproduce this too, also with 2013.

**Comment by fernando-jose-silva** [comment]
I opened the search window, pressed alt + tab, and pressed alt + tab again for the search window.
Nvda announces only localization of the dialog, not the name of the control that would be the search box.
pressing the left or right arrow key to navigate the character will mute the nvda.
log:
[nvda_excel1.txt](https://github.com/nvaccess/nvda/files/2834050/nvda_excel1.txt)

**Comment by lukaszgo1** [comment]
@feerrenrut Two more tests please:
1. Can you try with all add-ons disabled?
2. After opening find dialog press tab and the shift+tab and the try moving with the arrow keys.

**Comment by Adriani90** [comment]
Fernando could you please uninstall global plugins extensions addon and try again?
Also could you please go to nvda menu / tools / about nvda and copy the exact version of alpha you are using? Thanks.

Von meinem iPhone gesendet

**Comment by Adriani90** [comment]
Sorry U meant nvda menu /help / about nvda :)

Von meinem iPhone gesendet

**Comment by fernando-jose-silva** [comment]
I repeated the tests with the plug-ins disabled.
The results are the same, when walking character by character with the arrows the nvda is muted.
what the dialogue on the nvda says is:
[Window Title]
About NVDA

[Content]
NonVisual Desktop Access (NVDA)
Version: alpha-16682, b65f5bb7
Page: http://www.nvaccess.org/
Copyright (C) 2006-2018 NVDA Collaborators

NVDA is under the GNU General Public License (Version 2). You can distribute and modify the program as you wish, as long as the same license goes with it and offers the source code to whomever you wish. This applies to the original program and modified copies of it, as well as any derivative works.
For more details, see the license in the Help menu.
You can also see it at this address: https://www.magnux.org/doc/GPL-en.html

NVDA is developed by NV Access, a nonprofit organization committed to assisting and promoting free and open source solutions for blind and low vision people.
If NVDA is useful to you and wishes it to continue to improve, please consider donating to NV Access; you can do the Make a donation option in the NVDA menu.

[OK]

log:
[nvda_excel2.txt](https://github.com/nvaccess/nvda/files/2834486/nvda_excel2.txt)

**Comment by lukaszgo1** [comment]
@fernando-jose-silva Have you tried tabbing and then shift tabbing back to the find box?

**Comment by fernando-jose-silva** [comment]
I discovered the problem, there were 2 add-ons called excel and ospeak comands.
Now the problem is that when you press the command to open the search window nvda does not recognize the search field, it is necessary to press tab and then shift + tab to have the search field recognized and working correctly.
Thank you very much.

**Comment by leonardder** [comment]
I'm closing this, since the initial problem was solved in #9063. The problem that the edit field doesn't receive focus properly is a separate issue. Note that Narrator also suffers from this, so opening an issue for this will most likely result in a "needs external fix" label being added.
